telemetry/otelgrpc/otelgrpc.go

```diff
@@ -0,0 +1,198 @@
+package otelgrpc
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/goto/salt/utils"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/protobuf/proto"
+)
+
+type UnaryParams struct {
+	Start  time.Time
+	Method string
+	Req    any
+	Res    any
+	Err    error
+}
+
+type Monitor struct {
+	duration     metric.Int64Histogram
+	requestSize  metric.Int64Histogram
+	responseSize metric.Int64Histogram
+	attributes   []attribute.KeyValue
+}
+
+func NewOtelGRPCMonitor(hostName string) Monitor {
+	meter := otel.Meter("github.com/goto/salt/telemetry/otelgrpc")
+
+	duration, err := meter.Int64Histogram("rpc.client.duration", metric.WithUnit("ms"))
+	handleOtelErr(err)
+
+	requestSize, err := meter.Int64Histogram("rpc.client.request.size", metric.WithUnit("By"))
+	handleOtelErr(err)
+
+	responseSize, err := meter.Int64Histogram("rpc.client.response.size", metric.WithUnit("By"))
+	handleOtelErr(err)
+
+	addr, port := ExtractAddress(hostName)
+
+	return Monitor{
+		duration:     duration,
+		requestSize:  requestSize,
+		responseSize: responseSize,
+		attributes: []attribute.KeyValue{
+			semconv.RPCSystemGRPC,
+			attribute.String("network.transport", "tcp"),
+			attribute.String("server.address", addr),
+			attribute.String("server.port", port),
+		},
+	}
+}
+
+func GetProtoSize(p any) int {
+	if p == nil {
+		return 0
+	}
+
+	size := proto.Size(p.(proto.Message))
+	return size
```

Comment:
should we make it safer?
```suggestion
        if pm, ok := p.(proto.Message); ok {
            return proto.Size(pm)
        }
	return 0
```